From ba216731cd5bf4264e9ad95d55616ff1a9edfac5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Niklas=20Hamb=C3=BCchen?= <mail@nh2.me>
Date: Wed, 29 Aug 2018 12:31:04 +0200
Subject: [PATCH] Correctly declare libyaml dependency on system-libyaml flag.

Fixes

    stack build --extra-include-dirs ... --extra-lib-dirs ...

to set libyaml path being ignored.

Also fixes https://github.com/NixOS/cabal2nix/issues/372#issuecomment-416062945
---
 package.yaml | 25 ++++++++++++++-----------
 1 file changed, 14 insertions(+), 11 deletions(-)

Index: b/yaml.cabal
===================================================================
--- a/yaml.cabal
+++ b/yaml.cabal
@@ -113,6 +113,8 @@ library
         libyaml/reader.c
         libyaml/scanner.c
         libyaml/writer.c
+  else
+    extra-libraries: yaml
   if os(windows)
     cpp-options: -DWINDOWS
   default-language: Haskell2010
